ParquetClassLibrary/Sandbox/Room.cs

```diff
@@ -0,0 +1,128 @@
+        }
+
+        /// <summary>
+        /// The location with the least X and Y coordinates of every <see cref="Space"/> in this <see cref="Room"/>.
+        /// </summary>
+        /// <remarks>
+        /// A value of <c>null</c> indicates that this <see cref="Room"/> has yet to be located.
+        /// </remarks>
+        private Vector2Int? _cachedPosition;
+
+        /// <summary>
+        /// A location with the least X and Y coordinates of every <see cref="Space"/> in this <see cref="Room"/>.
+        /// </summary>
+        /// <remarks>
+        /// This location could server as a the upper, left point of a bounding rectangle entirely containing the room.
+        /// </remarks>
+        public Vector2Int Position
+            => (Vector2Int)(
+                null == _cachedPosition
+                    ? _cachedPosition = new Vector2Int(Perimeter.Select(space => space.Position.X).Min(),
+                                                       Perimeter.Select(space => space.Position.X).Min())
+                    : _cachedPosition);
+
+        /// <summary>
+        /// The cached <see cref="RoomRecipe"/> identifier.
+        /// A value of <see cref="EntityID.None"/> indicates that this <see cref="Room"/>
+        /// has yet to be matched with a <see cref="RoomRecipe"/>.
+        /// </summary>
+        private EntityID _cachedRecipeID = EntityID.None;
+
+        /// <summary>The <see cref="RoomRecipe"/> that this <see cref="Room"/> matches.</summary>
+        public EntityID RecipeID
+            => _cachedRecipeID == EntityID.None
+                ? _cachedRecipeID = All.Recipes.Rooms.FindBestMatch(this)
+                : _cachedRecipeID;
+
+        #region Initialization
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Room"/> class.
+        /// </summary>
+        /// <param name="in_walkableArea">
+        /// The <see cref="Space"/>s on which a <see cref="Characters.Being"/>
+        /// may walk within this <see cref="Room"/>.
+        /// </param>
+        /// <param name="in_perimeter">
+        /// The <see cref="Space"/>s whose <see cref="Block"/>s and <see cref="Furnishing"/>s
+        /// define the limits of this <see cref="Room"/>.
+        /// </param>
+        public Room(HashSet<Space> in_walkableArea, HashSet<Space> in_perimeter)
+        {
+            if (null == in_walkableArea)
+            {
+                throw new ArgumentNullException(nameof(in_walkableArea));
+            }
+            if (in_walkableArea.Count < All.Recipes.Rooms.MinWalkableSpaces
+                || in_walkableArea.Count > All.Recipes.Rooms.MaxWalkableSpaces)
+            {
+                throw new IndexOutOfRangeException(nameof(in_walkableArea));
+            }
+            if (null == in_perimeter)
+            {
+                throw new ArgumentNullException(nameof(in_perimeter));
+            }
+            var minimumPossiblePerimeterLength = (2 * in_walkableArea.Count) + 2;
```

Comment:
The amount of procedural attempts that could lead to exceptions makes me think that this might be better as a Factory, but for now these mostly seem like reasonable constraints. E.G. a room couldn't possibly have fewer perimeters than this, so if it ended up here it's a bad.

Reply:
Mm, that's an interesting thought.
I tend to avoid Factories but I am not sure I do that for any good reason.  I will consider this!